pkg/netrc/reader.go

```diff
@@ -0,0 +1,134 @@
+package netrc
+
+import (
+	"errors"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+var (
+	netrcOnce sync.Once
+	netrc     []netrcLine
+	netrcErr  error
+)
+
+// ReadNetrcPassword retrieves password for the desired server and login.
+// We just borrowed original code due to its internal nature, it is original implementation of .netrc read in Go
+// https://github.com/golang/go/blob/master/src/cmd/go/internal/auth/netrc.go
+func ReadNetrcPassword(jiraServer string, login string) (string, error) {
+	netrcOnce.Do(func() {
+		readNetrc()
+	})
+	if netrcErr != nil {
+		return "", netrcErr
+	}
+
+	jiraServerURL, err := url.ParseRequestURI(jiraServer)
```

Comment:
netrc has nothing to do with Jira and `netrc` package should not care anything about it at all.

I would suggest to keep `reader` and `reader_test` sync with the [original implementation](https://github.com/golang/go/blob/master/src/cmd/go/internal/auth/netrc.go) including license information (required as per BSD).

You can have a new file called `netrc.go` in the package that will expose required methods to work with `netrc` package.

```go
// Package netrc implements GNU .netrc specification.
// This implementation is borrowed from the original implementation by the go authors.
// See https://github.com/golang/go/blob/master/src/cmd/go/internal/auth/netrc.go
// See https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html
package netrc

import (
	"fmt"
)

// ErrNetRCEntryNotFound is thrown if details for the machine is not found.
var ErrNetRCEntryNotFound = fmt.Errorf("netrc: entry not found in config")

// Entry is a netrc config entry.
type Entry struct {
	Machine  string
	Login    string
	Password string
}

// Read reads config for the given machine.
func Read(machine string) (*Entry, error) {
	netrcOnce.Do(func() {
		readNetrc()
	})
	if netrcErr != nil {
		return nil, fmt.Errorf("netrc: %w", netrcErr)
	}

	for _, line := range netrc {
		if line.machine == machine {
			return &Entry{
				Machine:  line.machine,
				Login:    line.login,
				Password: line.password,
			}, nil
		}
	}

	return nil, ErrNetRCEntryNotFound
}
```

Now, you can use the exposed method elsewhere.

```go
netrcEntry, err := netrc.Read(config.Server)
```

Please feel free to update above implementation as necessary.